apps/studio/components/interfaces/Auth/Policies/AIPolicyEditorPanel/index.tsx

```diff
@@ -97,23 +80,26 @@ export const AIPolicyEditorPanel = memo(function ({
 
   const entityDefinitions = entities?.map((def) => def.sql.trim())
 
-  const { mutate: addPromptMutation } = useRlsSuggestMutation({
-    onSuccess: (data) => {
-      setIds({ threadId: data.threadId, runId: data.runId })
-    },
-    onError: (error) => {
-      const threadMessage = generateThreadMessage({
-        threadId: ids?.threadId,
-        runId: ids?.runId,
-        content: error.message.includes('No OPENAI_KEY set')
-          ? `Seems like you haven't set an OPENAI_KEY in your environment variables of the dashboard. Update your .env file with that environment variable to use all the AI features of the dashboard!`
-          : error.message,
-      })
-      setDebugThread([...debugThread, threadMessage])
-      setLoading(false)
+  const {
+    messages: chatMessages,
+    append,
+    isLoading,
+  } = useChat({
+    id: chatId,
+    api: `${BASE_PATH}/api/ai/sql/suggest`,
+    body: {
+      entityDefinitions: isOptedInToAI ? entityDefinitions : undefined,
+      policyDefinition:
+        selectedPolicy !== undefined ? generatePolicyDefinition(selectedPolicy) : undefined,
     },
   })
 
+  const messages = useMemo(() => {
+    const merged = [...debugThread, ...chatMessages.map((m) => ({ ...m, isDebug: false }))]
+
+    return sortBy(merged, (m) => m.createdAt)
```

Comment:
alaister ran into some scenarios where by the createdAt from the user and assistant were the exact same timestamps, leading to messages rendering in the wrong order sometimes

e.g 
user
assistant
assistant
user

our fix previously was to also sort on role if the created at is the same - but do you reckon we'll run into the same issue with streaming?

Reply:
Most probably, adding that trick here also.